### PR TITLE
Add IAM user resource

### PR DESCRIPTION
**Pull request by ds0440**
Added IAM user resource so that I did not have to work with it as a default resource type. You will notice I did add a `set_properties` method. I did this so I could load the IAM user properties specifically to be used for the `has_hardcoded_credentials` check. I wanted to check the `LoginProfile` property and receive a boolean value depending on if credentials were found. I thought this might work nicely with the method you already had. Let me know if you want to change they way this data is loaded. I bumped the version! Not sure if you want to just bump this to `0.3.1`. Technically if someone had automation written using the generic resource for an IAM user the change to a specific resource might break some things. The issue I ran into was with the generic resource the `policies` attribute returned the json loaded policies section of the CF template. Now that IAM user is a real resource the policies attribute will return a list of `pycfmodel.model.resources.properties.policy.Policy` objects. As mentioned let me know if you want me to change something.

### CODE REVIEW

**Comment by vlttnv** [comment]
Hi @ds0440 . As of this week I no longer work at Skyscanner so I will not be able to review/accept your PRs.
@oscarbc96 and the team can pick it up from here.
Good luck and thanks for your contributions!

File: pycfmodel/model/resources/iam_user.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        "UserName": String,
+        "ManagedPolicyArns": [ String, ... ],
+        "Path": String,
+        "Policies": [ Policies, ... ]
+        """
+        super().__init__(logical_id, value)
+
+        self.user_name = None
+        self.path = None
+        self.policies = []
```
[the remark this answers is not in the document]
Comment (ds0440) [reply pycfmodel/model/resources/iam_user.py]: Hey @oscarbc96! Good call. I will remove that initialization and and swap the default for an empty list instead of a dictionary. Not sure why I had that there.

File: pycfmodel/model/resources/iam_user.py [hunk 2]
```diff
@@ -28,13 +28,12 @@ def __init__(self, logical_id, value):
 
         self.user_name = None
         self.path = None
-        self.policies = []
 
         self.policies = self.get_policies(
-            value.get("Properties", {}).get("Policies"),
+            value.get("Properties", []).get("Policies"),
```
Comment (oscarbc96) [inline pycfmodel/model/resources/iam_user.py]: Hey! Maybe I'm wrong, but why did you change this from a dict to a list? Btw, a list doesn't have a `get` method.

File: pycfmodel/model/resources/iam_user.py [hunk 3]
```diff
@@ -28,13 +28,12 @@ def __init__(self, logical_id, value):
...
         )
         self.managed_policy_arns = self.get_managed_policy_arns(
-            value.get("Properties", {}).get("ManagedPolicyArns"),
+            value.get("Properties", []).get("ManagedPolicyArns"),
```
Comment (oscarbc96) [inline pycfmodel/model/resources/iam_user.py]: Same as before.

File: pycfmodel/model/resources/iam_user.py (hunk 2 above)
Comment (ds0440) [reply pycfmodel/model/resources/iam_user.py]: Oh my, because I was not paying attention. My intent was to have the default of getting `value['Properties']['Policies']` equal and empty list if there are no polices or the `Policies` key is not present. Previously there were a handful of issues where parts of the code were expecting lists  and received strings or other. I will update the code. Sorry let me know if you see any other issues.

**Comment by ds0440** [comment]
I see why I was initializing policies as an empty list. I think I duplicated the IAM role resource without looking into it enough. That resource also has the same initialization and it appears as though it will be overwritten by the result of `get_policies` in the same way. Not sure if I should fix that here. Might be better for that to happen in a separate pull. Thoughts?

**Comment by oscarbc96** [comment]
I would create a new PR to solve it, and then we can merge this one. What do you think?

**Comment by ds0440** [comment]
Yeah that works for me. If I am understanding you correctly I should first make a new branch to fix the improper/unneeded initialization issues and bump the version to `0.3.1`. Once that PR is merged bump the version for this PR to `0.3.2`? I also did want to make sure that you saw the above comment about the change in behavior from a generic resource to a defined resource and that it could potentially break anything previously written for IAM users as a generic resource. I am bringing this up because I want to make sure I am bumping to a version that makes sense for the type of change. I am good with whatever you guys want, I just never heard back before. Thanks!

**Comment by ds0440** [comment]
I merged the changes from commit #17 into this pull request and bumped the version as well
